pkg/service/service.go

```diff
@@ -785,19 +785,18 @@ func (s *Service) upsertServiceIntoLBMaps(svc *svcInfo, onlyLocalBackends bool,
 	}
 
 	// Upsert service entries into BPF maps
-	backends := make(map[string]lb.BackendID, len(svc.backends))
+	backends := make(map[string]*lb.Backend, len(svc.backends))
 	activeBackendsCount := 0
-	for _, b := range svc.backends {
-		// Skip adding the terminating backend (or backend with weight 0)
-		// to the service map so that it won't be selected to serve new requests.
-		// However, the backend is still
+	for idx, b := range svc.backends {
+		// Skip adding the terminating backend to the service map so that it
+		// won't be selected to serve new requests. However, the backend is still
 		// kept in the affinity and backend maps so that existing connections
 		// are able to terminate gracefully. The final clean-up for the backend
 		// will happen once the agent receives a delete event for the service
 		// endpoint as it'll be part of obsoleteBackendIDs/obsoleteSVCBackendIDs
 		// list passed to this function.
-		if !b.Terminating && b.Weight > 0 {
```

Comment:
Why is this getting removed?

Reply:
the same as the comment above: I realized then I need to move the logic closer to maglev table population